nixos/modules/services/web-apps/libretranslate.nix

```
@@ -0,0 +1,203 @@
+        default = null;
+        example = 8;
+      };
+      enableApiKeys = lib.mkOption {
+        description = lib.mdDoc "Enable API keys database.";
+        type = lib.types.bool;
+        default = false;
+        example = true;
+      };
+      disableWebUI = lib.mkOption {
+        description = lib.mdDoc "Disable web ui.";
+        type = lib.types.bool;
+        default = false;
+        example = true;
+      };
+      updateModels = lib.mkOption {
+        description = lib.mdDoc "Update language models at startup";
+        type = lib.types.bool;
+        default = false;
+        example = true;
+      };
+      domain = lib.mkOption {
+        description = lib.mdDoc "The domain serving your LibreTranslate instance. Required for configure nginx as a reverse proxy.";
+        type = lib.types.str;
+        default = "";
+        example = "libretranslate.example.com";
+      };
+      configureNginx = lib.mkOption {
+        description = lib.mdDoc "Configure nginx as a reverse proxy for LibreTranslate.";
+        type = lib.types.bool;
+        default = false;
+      };
+      extraArgs = lib.mkOption {
+        description = lib.mdDoc "Extra arguments passed to the LibreTranslate.";
+        type = lib.types.listOf lib.types.str;
+        default = [ ];
+        example = [
+          "--debug"
+          "--disable-files-translation"
+        ];
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ ltmanageKeysCli ];
+
+    systemd.tmpfiles.rules = lib.mkIf (cfg.dataDir != "/var/lib/libretranslate") [
+      "d '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+      "z '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.libretranslate = {
+      description = "LibreTranslate service";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      environment = {
+        HOME = cfg.dataDir;
+      };
+      serviceConfig = lib.mkMerge [{
+        Type = "simple";
+        ExecStart = ''
+          ${cfg.package}/bin/libretranslate \
+            --host ${cfg.host} \
+            --port ${toString cfg.port} \
+            ${lib.optionalString cfg.enableApiKeys "--api-keys"} \
+            ${lib.optionalString cfg.disableWebUI "--disable-web-ui"} \
+            ${lib.optionalString cfg.updateModels "--update-models"} \
+            ${lib.optionalString (cfg.threads != null) "--threads ${toString cfg.threads}"} \
+            ${lib.concatStringsSep " " cfg.extraArgs}
+        '';
+        WorkingDirectory = cfg.dataDir;
+        # User and group
+        User = cfg.user;
+        Group = cfg.group;
+        # Proc filesystem
+        ProcSubset = "all";
+        ProtectProc = "invisible";
+        # Access write directories
+        UMask = "0027";
+        # Capabilities
+        CapabilityBoundingSet = "";
+        # Security
+        NoNewPrivileges = true;
+        # Sandboxing
```

```suggestion
        ProcSubset = "all";
        ProtectProc = "invisible";
        UMask = "0027";
        CapabilityBoundingSet = "";
        NoNewPrivileges = true;
```
those comments don't add much and all those settings are sandboxing and security related.